### PR TITLE
Create proposal_RooUnfold.md

### PR DESCRIPTION
**Pull request by vincecr0ft**
adding the RooUnfold proposal on behalf of the RooUnfold and RooFit development teams.

### CODE REVIEW

File: _gsocproposals/2021/proposal_RooUnfold.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+---
+
+## Description
+[RooUnfold](https://roounfold.web.cern.ch) is a powerful modular analysis tool is for the unfolding (AKA "deconvolution" or "unsmearing") of distributions.
+Unfolding is the statistical process in which the 'true' underlying processes that govern the interactions of fundamental forces of the universe are estimated from both simulated and measured data.
+To facilitate these data intensive analyses, RooUnfold is built on the [ROOT](https://root.cern/) Software Framework and is used in the publication of hundreds of LHC analyses every year.
+RooUnfold was designed to provide a common interface for a number of analysis algorithms and was [recently upgraded](https://arxiv.org/abs/1910.14654) to leverage the power of the ROOT statistical modelling package RooFit in order to provide a statistical comparison of the various algorithms.
+
+This project aims to address obscelescance in the underlying RooUnfold implementation that would allow the RooUnfold package to act as a lightweight member class to ROOT whilst retaining it’s functionality and the ability for users to define new algorithms including machine learning based solutions, and data intensive tasks.
```
Comment (aflp91) [inline _gsocproposals/2021/proposal_RooUnfold.md:17]: obscelescance => obsolescence ?
retaining it’s functionality => retaining its functionality ?

File: _gsocproposals/2021/proposal_RooUnfold.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+This project aims to address obscelescance in the underlying RooUnfold implementation that would allow the RooUnfold package to act as a lightweight member class to ROOT whilst retaining it’s functionality and the ability for users to define new algorithms including machine learning based solutions, and data intensive tasks. 
+
+## Task Ideas
+* Learn about the algorithms and statistics that govern the LHC's most precise measurements.
+* Implement a lightweight RooUnfold base class using modern C++ that is able to efficiently replicate current usage.
+* Design and develop an interface for analysis algorithms that is compatible with the new array of statistical tests available.
+* Explore the implementation of deep learning based algorithms that leverage sparse weighted events rather than histograms.
+* Expand the suite of data visualisation tools and performance metrics provided for the unfolding proceedure.
```
Comment (aflp91) [inline _gsocproposals/2021/proposal_RooUnfold.md:24]: proceedure => procedure ?

**Review by aflp91 — CHANGES_REQUESTED** [review]
Vincent, could you please add mentors in `gsoc/2021/mentors.md` file?

File: _gsocproposals/2021/proposal_RooUnfold.md (hunk 2 above)
Comment (vincecr0ft) [reply _gsocproposals/2021/proposal_RooUnfold.md]: Fixed

File: _gsocproposals/2021/proposal_RooUnfold.md (hunk 1 above)
Comment (vincecr0ft) [reply _gsocproposals/2021/proposal_RooUnfold.md]: fixed

Comment (aflp91) [reply _gsocproposals/2021/proposal_RooUnfold.md]: Thank you, Vincent, but could you put entries sorted in last name alphabetic order?

Comment (vincecr0ft) [reply _gsocproposals/2021/proposal_RooUnfold.md]: ah ok yes. Very democratiic. I'll do that now.

Comment (vincecr0ft) [reply _gsocproposals/2021/proposal_RooUnfold.md]: in mentors.md also?

Comment (aflp91) [reply _gsocproposals/2021/proposal_RooUnfold.md]: Thanks! 
But 2 files are missing too:
- a file describing RooUnfold project (`_gsocprojects/2021/project_ RooUnfold.md`)
- a file describing Tufts organization (`_gsocorgs/2021/tufts.md`)

Comment (vincecr0ft) [reply _gsocproposals/2021/proposal_RooUnfold.md]: ok I think that's done! Thanks for all your effort on this!

**Review by aflp91 — CHANGES_REQUESTED** [review]
You need to add a "Tab" in front of the project description line.
Except that, it seems OK with me!

**Review by aflp91 — APPROVED** [review]
Thanks! OK with me.

**Pull request merged**